appserver/ejb/ejb-rest-remoting/admin/src/main/java/fish/payara/ejb/rest/admin/EnableEjbInvokerCommand.java

```diff
@@ -0,0 +1,118 @@
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.ejb.rest.admin;
+
+import static org.glassfish.config.support.CommandTarget.CLUSTER;
+import static org.glassfish.config.support.CommandTarget.CLUSTERED_INSTANCE;
+import static org.glassfish.config.support.CommandTarget.CONFIG;
+import static org.glassfish.config.support.CommandTarget.DAS;
+import static org.glassfish.config.support.CommandTarget.STANDALONE_INSTANCE;
+
+import javax.inject.Inject;
+
+import org.glassfish.api.ActionReport;
+import org.glassfish.api.Param;
+import org.glassfish.api.admin.AdminCommand;
+import org.glassfish.api.admin.AdminCommandContext;
+import org.glassfish.api.admin.ExecuteOn;
+import org.glassfish.api.admin.RuntimeType;
+import org.glassfish.api.admin.ServerEnvironment;
+import org.glassfish.config.support.TargetType;
+import org.glassfish.deployment.autodeploy.AutoDeployer.AutodeploymentStatus;
+import org.glassfish.deployment.autodeploy.AutoDeploymentOperation;
+import org.glassfish.hk2.api.PerLookup;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.jvnet.hk2.annotations.Service;
+
+/**
+ * This command enables the EJB invoker endpoint and optionally allows it to set an
+ * alternative context root.
+ * 
+ * <p>
+ * This happens by the deploying a small application in <code>/domains/[domain]/endpoints/ejb-invoker</code>
+ * which by default is not deployed. The default context root of this application is 
+ * <code>/ejb-invoker</code>.
+ * 
+ * @author Arjan Tijms
+ *
+ */
+@Service(name = "enable-ejb-invoker")
+@PerLookup
+@ExecuteOn(RuntimeType.DAS)
+@TargetType(value = { DAS, STANDALONE_INSTANCE, CLUSTER, CLUSTERED_INSTANCE, CONFIG })
+public class EnableEjbInvokerCommand implements AdminCommand {
+
+    private final String ENDPOINT = "endpoints/ejb-invoker";
+    
+    @Param(name = "contextRoot", primary = true, optional = true)
+    private String contextRoot;
+    
+    @Inject
+    private ServerEnvironment serverEnvironment;
+    
+    @Inject
+    private ServiceLocator serviceLocator;
+
+    @Override
+    public void execute(AdminCommandContext context) {
+        AutoDeploymentOperation autoDeploymentOperation = AutoDeploymentOperation.newInstance(
+                serviceLocator,
+                serverEnvironment.getInstanceRoot().toPath().resolve(ENDPOINT).toFile(), 
+                getDefaultVirtualServer(),
+                getTarget(),
+                contextRoot
+                );
+        
+        AutodeploymentStatus deploymentStatus = autoDeploymentOperation.run();
+        
+        ActionReport report = context.getActionReport();
+        report.setActionExitCode(deploymentStatus.getExitCode());
+    }
+    
+    private String getDefaultVirtualServer() {
```

Comment:
The status of this is unclear to me. Should it throw an `UnsupportedOperationException`?

Reply:
No, the method is only called internally (it's a private method). Basically, it's a fancy `null`. The auto deploy operation that is uses has a very similar method.